tests/test_read.py

```diff
@@ -62,7 +62,7 @@ def btl_duplicate_header_name():
 def btl_as_stream():
```

Comment:
This does not need to be a fixture b/c we won't be calling it multiple times in tests.
You can remove the  `@pytest.fixture` and rename it to `test_btl_as_stream` so pytest can execute it.

Reply:
Sorry, I'm not use to unittesting specifically with pytest I was just emulating what other tests had done, I'll fix it up.

Reply:
Don't worry and no need to be sorry. The only reason I'm not sending commit directly to your branch is b/c I believe you want to do this. If not, I can take over. Your contributions and very welcomed and I'm trying to pay back by showing a bit how my messy tests work.